### DISCUSSION TITLE
Google OAuth2 API responds with a null body but with status code 200 - OK

### DISCUSSION THREAD

**Discussion by dmdum** [googleapis/googleapis#743] [Q&A] [answered]
I'm trying to fetch an access token from https://oauth2.googleapis.com/token. I'm getting a status code of 200 but the body is null. 

Logging the response with the code below:
`
log.debug("RESULT BODY: {} {} {}", result.code(), result.reason(), result.body());
`

Outputs this:
`RESULT BODY: 200 OK null`

I'm not exactly sure what's wrong here. Any ideas or reading materials you have that might help is greatly appreciated. Thanks!

**Comment by dmdum** [comment]
Hi @garrettjonesgoogle , sorry to blatantly tag you like this but I saw you were a member of the repo and was wondering if you have any idea why this happens? Couldn't find a similar case on the internet yet.

**Reply by Matti-gitdev** [reply]
Hello

Your token web address not work error 404


https://www.ionos.com/digitalguide/websites/website-creation/what-does-the-404-not-found-error-mean/

**Reply by dmdum** [reply]
Hello, the address was provided by Google here: https://developers.google.com/identity/protocols/oauth2/web-server#exchange-authorization-code

**Reply by dmdum** [reply]
I traced the http request and found out that the API was responding with a body but it wasn't properly mapped to the variable containing the result.

**Comment by dmdum** [comment, marked as answer]
By tracing the http request, I found out that the API was responding with a body but it wasn't properly mapped to the variable containing the result which is why it was showing as null.
